src/RefractDataStructure.cc

```diff
@@ -514,51 +663,69 @@ namespace drafter {
         size_t valuesCount = data.values.size();
 
         if (!data.descriptions.empty()) {
-            element->meta[SerializeKey::Description] = refract::IElement::Create(*data.descriptions.begin());
+            element->meta[SerializeKey::Description] = DescriptionToRefract(data);
         }
 
-        SetElementType(element, value.valueDefinition.typeDefinition);
+        SetElementType(element, value.section.valueDefinition.typeDefinition);
 
-        std::for_each(value.sections.begin(), value.sections.end(), ExtractTypeSection<T>(data, value));
+        SectionInfoCollection<mson::TypeSections> typeSections(value.section.sections, value.sourceMap.sections);
+
+        std::for_each(typeSections.begin(), typeSections.end(), ExtractTypeSection<T>(data, value));
         
-        if (!value.valueDefinition.values.empty() && (valuesCount != data.values.size())) { 
+        if (!value.section.valueDefinition.values.empty() && (valuesCount != data.values.size())) { 
             // there are some values coming from TypeSections -> move first value into examples
             ElementType* element = new ElementType;
             element->set(data.values.front());
             data.samples.insert(data.samples.begin(), element);
             data.values.erase(data.values.begin());
+
+            // FIXME append source map into "sample"
+            data.valuesSourceMap.erase(data.valuesSourceMap.begin());
         }
 
-        TransformElementData(element, data);
+        TransformElementData(element, data, value.hasSourceMap());
 
         return element;
     }
 
     template <typename T>
-    refract::MemberElement* RefractElementFromProperty(const mson::PropertyMember& property, const mson::BaseTypeName defaultNestedType)
+    refract::MemberElement* RefractElementFromProperty(const SectionInfo<mson::PropertyMember>& property, const mson::BaseTypeName defaultNestedType)
     {
         refract::MemberElement* element = new refract::MemberElement;
-        refract::IElement* value = RefractElementFromValue<T>(property, defaultNestedType);
+        refract::IElement* value = RefractElementFromValue<T>(SectionInfo<mson::ValueMember>(property.section, property.sourceMap), defaultNestedType);
```

Comment:
No I can't because there is downcast from `mson::PropertyMember` to `mson::ValueMember`

For same reason I can not use `property' itself.